alacritty/src/ipc.rs

```diff
@@ -40,13 +45,39 @@ pub fn spawn_ipc_socket(options: &Options, event_proxy: EventLoopProxy<Event>) -
 
     // Spawn a thread to listen on the IPC socket.
     thread::spawn_named("socket listener", move || {
-        // Accept up to 2 bytes to ensure only one byte is received.
-        // This ensures forward-compatibility.
-        let mut buf = [0; 2];
-
-        while let Ok(received) = socket.recv(&mut buf) {
-            if buf[..received] == SOCKET_MESSAGE_CREATE_WINDOW {
-                let _ = event_proxy.send_event(Event::new(EventType::CreateWindow, None));
+        let mut data = Vec::new();
+        for stream in listener.incoming() {
+            let stream = match stream {
+                Ok(stream) => stream,
+                Err(err) => {
+                    warn!("Failed to get incomming stream! {}", err);
+                    continue;
+                },
+            };
+
+            data.clear();
+            let mut stream = BufReader::new(stream);
+
+            match stream.read_until(JSON_SEPARATOR, &mut data) {
+                Ok(0) | Err(_) => continue,
+                Ok(bytes_read) => bytes_read,
+            };
+
+            // Read pending events on socket.
+            let message: IPCMessage = match serde_json::from_slice(&data) {
+                Ok(message) => message,
+                Err(err) => {
+                    warn!("Fail to get data from socket: {}", err);
```

Comment:
```suggestion
                    warn!("Fail to convert data from socket: {}", err);
```

I think convert is better than `get` here, since we were able to get the data from the socket the only thing that didn't work is parsing it.

Reply:
agree.